README.md

```diff
@@ -1,3 +1,137 @@
+   2. An internal corporate mirror can be 100x faster than a public one, especially for CI builds.
+
+`gradle-jdks` solves all of these problems:
+
+1. **You can choose your favoured JDK vendor.**
+2. **Use the same JDK version on dev machines, in CI and in prod.**
+3. **Automatically add JDK CA certificates.**
+4. **Point to an internal mirror for JDKs.**
+
+## Usage
+
+_**Palantirians:** you probably shouldn't use this plugin directly - either use [`gradle-jdks-latest`](https://github.com/palantir/gradle-jdks-latest) for OSS or `gradle-jdks-internal` for internal projects which will set everything up for you._
+
+First, you must apply the plugin. In the **root project**, Either use the new plugin syntax:
+
+```gradle
+plugins {
+   id 'com.palantir.jdks' version '<latest version>'
+}
+```
+
+or the old buildscript method:
+
+```gradle
+buildscript {
+   repositories {
+      mavenCentral()
+   }
+   
+   dependencies {
+      classpath 'com.palantir.gradle.jdks:gradle-jdks:<latest-version>'
+   }
+}
+
+apply plugin: 'com.palantir.jdks'
+```
+
+Next up comes configuring the JDKs plugin:
+
+```gradle
+jdks {
+   // Required: For each Java major version you use, you need to specify
+   //           a distribution and version.
+   jdk(11) {
+      distribution = 'azul-zulu'
+      jdkVersion = '11.54.25-11.0.14.1'
+   }
+   
+   jdk(17) {
+      distribution = 'amazon-corretto'
+      jdkVersion = '17.0.3.6.1'
+   }
+   
+   // Optional: For each distribution, you can set a base url for a
+   //           mirror to use instead of the default public mirror.
+   // Default:  Whatever mirror is publicly provided by the vendor
+   jdkDistribution('azul-zulu') {
+      baseUrl = 'https://internal-corporate-mirror/azul-zulu-cdn-mirror'
+   }
+   
+   // Optional: You can specify CA certs which will be installed into
+   //           the extracted JDK to work with TLS interception.
+   // Default:  No CA certs are added.
+   caCerts.put 'corporate-tls-cert', '''
+      -----BEGIN CERTIFICATE-----
+      // snip
+      -----END CERTIFICATE-----
+   '''.stripIndent(true)
+   
+   // Optional: Where to store the JDKs on disk. You almost certainly
+   //           do not need to change this. 
+   // Default:  $HOME/.gradle/gradle-jdks
+   jdkStorageLocation = System.getProperty("user.home") + '/custom/location'
+}
+```
+
+Behind the scenes, `gradle-jdks` applies [`com.palantir.baseline-java-versions` (another gradle plugin - more docs in link)](https://github.com/palantir/gradle-baseline#compalantirbaseline-java-versions) to handle configuring the Java language versions. **You will need to configure this plugin as well** to tell it what Java language versions:
+
+Whilst eating food,  
```

Comment:
?

Reply:
When you're hungry while writing docs

Reply:
Bahaha - I got annoyed at IntelliJ telling me `In future` was bad english because it's British English and tried out `whilst` to check it's not being totally dumb. Must have committed it!

<img width="446" alt="Screenshot 2022-06-09 at 17 35 26" src="https://user-images.githubusercontent.com/397795/173054921-ae53d198-3807-478b-bf12-ffd33d4eeda6.png">